frontend/src/components/eventForm/eventForm.js

```diff
@@ -27,7 +27,7 @@ const inputField = ({
 		variant={variant} {...input}
 		placeholder={placeholder} helperText={helperText}
 		children={children} required={required}
-    fullWidth onChange={onChange}
+    fullWidth onBlur={onChange}
```

Comment:
I understand the need for a custom input element and why you're creating a material-ui wrapper to avoid repeating the props mapping in every formik `<Field>`. You could still eliminate some _WET_ code by using rest operator to pass props to `<TextField>` instead of destructuring paeed parameters then repeating yourself twice, like so:

WET:
```
const inputField = ({
  input,
  children,
  id,
  label,
  type,
  variant,
  placeholder,
  onChange,
  required,
  helperText,
  margin,
  defaultValue,
  multiline,
  rows
}) => (
  <TextField
    id={id}
    label={label}
    type={type}
    variant={variant}
    {...input}
    placeholder={placeholder}
    helperText={helperText}
    children={children}
    required={required}
    fullWidth
    onChange={onChange}
    margin={margin}
    defaultValue={defaultValue}
    multiline={multiline}
    rows={rows}
  />
);
```

DRY:
```
const inputField = ({
  input,
  ...rest
}) => (
  <TextField
    {...input}
    {...rest}
    fullWidth
  />
);
```
and of course if there are any nested props you could extract them first then unpack the rest in the wrapped component.

Reply:
There are several cases in UI code that follow a similar pattern– which may require very little effort to refactor yet would significantly improve code readability and maintainability

Reply:
Also please consider using a linter & a formatter as I noticed inconsistent indentation across different patterns of the code, I recommend ESLint/Prettier combo if you use VSCode

Reply:
Needless to say, If you decide to spread props/JSX attributes in the wrapped component, you need to make sure the the correct mapping is set in `<Field>` instance. So this hack does NOT work :smile:
   
```suggestion
    fullWidth onBlur={OnChange}
```

Reply:
On a side note, I suggest you make use of functional components to reduce duplicate/boilerplate code. For example, you could replace 3 `inputField` definitions in `login.js`, `register.js` and `eventForm.js` with a single reusable `<InputField>` that would be used across different forms